pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -159,39 +159,6 @@ func wrongTypeParamsNames(params []v1beta1.Param, matrix []v1beta1.Param, needed
 	return wrongTypeParamNames
 }
 
-// MissingKeysObjectParamNames checks if all required keys of object type params are provided in taskrun params or taskSpec's default.
-func MissingKeysObjectParamNames(paramSpecs []v1beta1.ParamSpec, params []v1beta1.Param) map[string][]string {
-	neededKeys := make(map[string][]string)
-	providedKeys := make(map[string][]string)
-
-	for _, spec := range paramSpecs {
-		if spec.Type == v1beta1.ParamTypeObject {
-			// collect required keys from properties section
-			for key := range spec.Properties {
-				neededKeys[spec.Name] = append(neededKeys[spec.Name], key)
-			}
-
-			// collect provided keys from default
-			if spec.Default != nil && spec.Default.ObjectVal != nil {
-				for key := range spec.Default.ObjectVal {
-					providedKeys[spec.Name] = append(providedKeys[spec.Name], key)
-				}
-			}
-		}
-	}
-
-	// collect provided keys from run level value
-	for _, p := range params {
-		if p.Value.Type == v1beta1.ParamTypeObject {
-			for key := range p.Value.ObjectVal {
-				providedKeys[p.Name] = append(providedKeys[p.Name], key)
-			}
-		}
-	}
-
-	return findMissingKeys(neededKeys, providedKeys)
-}
-
 // findMissingKeys checks if objects have missing keys in its providers (taskrun value and default)
 func findMissingKeys(neededKeys, providedKeys map[string][]string) map[string][]string {
```

Comment:
is this dead code now?

Reply:
no it is also used by other functions in that file. Other functions are not shared by taskrun and pipelinerun so I didn't move them into the new pkg

Reply:
I just realized the version in the `paramvalidation` package is different from the version here. Is this a behavior change? Also, the new function is just used by MissingKeysObjectParamNames as far as I can tell. It seems like this commit is mainly focused on array param validation-- maybe you could tackle validation related to object params in a separate PR?

Reply:
Okok

Reply:
The reason I moved the MissingKeysObjectParamNames is because it is also imported from taskrun pkg to pipelinerun's pkg.

Reply:
ah ok! This seems like a good thing to add a todo or issue for-- thanks for thinking to do the additional refactoring, although I think it's better to scope the PR just to the array indexing valiation.

Reply:
Thanks! I have a bunch of TODO waiting for this PR 😄